src/Agent.php

```diff
@@ -329,6 +329,10 @@ public function handleAgent($metadata)
             $input['tag'] = $metadata['tag'];
         }
 
+        if (isset($metadata['port']) && !is_null($metadata['port'])) {
```

Comment:
`isset(null)` returns `false`, no need to do the second check.
```suggestion
        if (isset($metadata['port'])) {
```